src/game/Objects/Player.cpp

```diff
@@ -1158,9 +1158,10 @@ struct UpdateAttackersCombatHelper
     explicit UpdateAttackersCombatHelper(Player* _player) : player(_player) {}
     void operator()(Unit* unit) const
     {
-        for (Unit* attacker : unit->getAttackers())
-            if (Creature* creature = attacker->ToCreature())
-                if (attacker->getVictim() == unit)
+        Unit::AttackerSet const attackers = unit->getAttackers();
+        for (Unit::AttackerSet::const_iterator itr = attackers.begin(); itr != attackers.end(); ++itr)
```

Comment:
Can I ask why the simpler for-loop syntax cause a crash?

Reply:
^ all this is is a syntax change, i assume there is some concurrent modification going on

Reply:
OK, I see

Reply:
The loop is slightly different by the use of const_iterator but that's not the point.
That is copying the structure first like it's done elsewhere in the core, I couldn't reproduce the crash afterwards.

Reply:
Maybe getAttackers contains a nullptr so it crashes when you call ToCreature?

You can replace:
`if (Creature* creature = (*itr)->ToCreature())`

with

`if (Creature* creature = (*itr) && (*itr)->IsCreature() ? (*itr)->ToCreature() : nullptr)`